tests/unit/services/date.spec.js

```diff
@@ -0,0 +1,138 @@
+    expect(sameDate.getSeconds()).toBe(0);
+    expect(sameDate.getMilliseconds()).toBe(0);
+
+    // expect to get 04.03.1988 back, after adding 5 days
+    addedDayes = 5;
+    const resetDate = getNDaysFromDate(fullDate, addedDayes);
+    expect(resetDate.getFullYear()).toBe(year);
+    expect(resetDate.getMonth()).toBe(2);
+    expect(resetDate.getDate()).toBe(4);
+    expect(resetDate.getHours()).toBe(0);
+    expect(resetDate.getMinutes()).toBe(0);
+    expect(resetDate.getSeconds()).toBe(0);
+    expect(resetDate.getMilliseconds()).toBe(0);
+
+    // expect to get 27.02.1988 back, after deducting a day
+    addedDayes = -1;
+    const yesterday = getNDaysFromDate(fullDate, addedDayes);
+    expect(yesterday.getFullYear()).toBe(year);
+    expect(yesterday.getMonth()).toBe(month);
+    expect(yesterday.getDate()).toBe(date - 1);
+    expect(yesterday.getHours()).toBe(0);
+    expect(yesterday.getMinutes()).toBe(0);
+    expect(yesterday.getSeconds()).toBe(0);
+    expect(yesterday.getMilliseconds()).toBe(0);
+  });
+
+  it('convertMsToDays converts ms to number of dats', () => {
+    const days = convertMsToDays(5 * 24 * 60 * 60 * 1000);
+    expect(days).toBe(5);
+  });
+
+  it(`getDaysDateRange returns array of dates between 2 dates,
+      including start date and excluding end date`, () => {
+    // we test for interval between fullDate(28.02.1988) and 05.03.1988
+    const futureDate = new Date(year, 2, 5, hour, minute, second, miliSecond);
+    const datesList = getDaysDateRange(fullDate, futureDate);
+    expect(datesList.length).toBe(6);
+  });
+
+  it(`'getIntlDateString takes an array of dates and returns a list of unique day names
+      in those interval`, () => {
+    const datesList = [
+      '1988-02-28T00:00:00.000Z',
+      '1988-02-29T00:00:00.000Z',
+      '1988-03-01T00:00:00.000Z',
+      '1988-03-02T00:00:00.000Z',
+      '1988-03-03T00:00:00.000Z',
+      '1988-03-04T00:00:00.000Z',
+      '1988-03-05T00:00:00.000Z',
+      '1988-03-06T00:00:00.000Z',
+      '1988-03-07T00:00:00.000Z',
+      '1988-03-08T00:00:00.000Z',
+      '1988-03-09T00:00:00.000Z',
+      '1988-03-10T00:00:00.000Z',
+      '1988-03-11T00:00:00.000Z',
+      '1988-03-12T00:00:00.000Z',
+      '1988-03-13T00:00:00.000Z',
+      '1988-03-14T00:00:00.000Z',
+      '1988-03-15T00:00:00.000Z',
+      '1988-03-16T00:00:00.000Z',
+      '1988-03-17T00:00:00.000Z',
+      '1988-03-18T00:00:00.000Z',
+      '1988-03-19T00:00:00.000Z',
+    ];
+    const daysInIntervalLong = getIntlDateString('en-US', datesList, { weekday: 'long' });
+    expect(daysInIntervalLong.length).toBe(7);
+    expect(daysInIntervalLong[0]).toBe('Sunday');
+
+    const daysInIntervalShort = getIntlDateString('en-US', datesList, { weekday: 'short' });
+    expect(daysInIntervalShort.length).toBe(7);
+    expect(daysInIntervalShort[0]).toBe('Sun');
+  });
+
+  it('generateRandomIndexes returns array of random integers on open interval', () => {
```

Comment:
I am not sure why is this part of date service? Should I move it somewhere else? 

Also there's something strange in design here, that I am unsure about. If you look (here)[https://github.com/homeday-de/homeday-blocks/blob/develop/src/services/date.js#L45-L56] - to make it easier to discuss, I'm copying the relevant part of the code: 

```javascript
export const generateRandomIndexes = (amount, min, max) => {
  const indexes = [];

  while (indexes.length < amount) {
    const randomInt = getRandomInt(min, max);
    /* eslint-disable no-continue  */
    if (indexes.indexOf(randomInt) > -1) continue;
    indexes.push(getRandomInt(min, max));
  }

  return indexes;
};
```

What is the purpose of the `continue` statement? Initially I've read this as making sure we are returning unique indexes, however that does not seem to be the case as at the end, we don't push our original index, but generate new random value -> `indexes.push(getRandomInt(min, max));` . Anyone has more insights about implementation?  

This function also has potential to block the main threat if input params are not selected carefully. 
e.g.: 
```javascript
const amount = 100;
const min = 1;
const max = 2;
generateRandomIndexes (count, min, max);
```

That would loop....should we add some sanity checks at the beginning? I can add it in this PR. 

Anyway, if someone has more insights on actual usage here, I'll be happy to update this and document properly.

Reply:
If I'm not mistaken this is used to mock some charts data in one of our projects.
I agree with you `continue` is useless here, `randomInt` should be pushed to the array.

IMO this function seems too specific, I don't think we should have it in Blocks, or at least it should go to [utils.js](https://github.com/homeday-de/homeday-blocks/blob/develop/src/services/utils.js)

Reply:
I think it fits `utils` more indeed. Probably renamed to `getNRandomNumbersOnTheInterval` or something shorter. :) 

I'll jobby it up and move it. Will also add sanity check to avoid issues.

Reply:
It is actually copied from this guy: https://github.com/homeday-de/sliders/blob/d1356d2e1200a794aaadf0ac73823513ca1210b3/src/services/dateService.js#L47

It is sliders specific and might have been copied as a whole. I'm completely fine about removing this guy from blocks and leave it for sliders.